src/Setup/DevDivVsix/CompilersPackage/Microsoft.CodeAnalysis.Compilers.swr

```diff
@@ -27,12 +27,12 @@ folder InstallDir:\MSBuild\15.0\Bin\Roslyn
     file source=$(OutputPath)\Dlls\MSBuildTask\net46\Microsoft.Managed.Core.targets
     file source=$(OutputPath)\Dlls\MSBuildTask\net46\Microsoft.VisualBasic.Core.targets
     file source=$(OutputPath)\Dlls\MSBuildTask\net46\Microsoft.CSharp.Core.targets
-
+    file source=$(OutputPath)\Dlls\MSBuildTask\net46\Microsoft.Build.Tasks.CodeAnalysis.dll vs.file.ngenArchitecture=all
+    
     file source=$(OutputPath)\Dlls\Scripting\Microsoft.CodeAnalysis.Scripting.dll vs.file.ngenArchitecture=all vs.file.ngenApplication="[installDir]\MSBuild\15.0\Bin\Roslyn\csi.exe"
     file source=$(OutputPath)\Dlls\CSharpScripting\Microsoft.CodeAnalysis.CSharp.Scripting.dll vs.file.ngenArchitecture=all vs.file.ngenApplication="[installDir]\MSBuild\15.0\Bin\Roslyn\csi.exe"
 
     file source=$(OutputPath)\Dlls\CSharpCodeAnalysis\Microsoft.CodeAnalysis.CSharp.dll vs.file.ngenArchitecture=all vs.file.ngenApplication="[installDir]\MSBuild\15.0\Bin\Roslyn\VBCSCompiler.exe"
-    file source=$(OutputPath)\Dlls\MSBuildTask\net46\Microsoft.Build.Tasks.CodeAnalysis.dll vs.file.ngenArchitecture=all vs.file.ngenApplication="[installDir]\MSBuild\15.0\Bin\Roslyn\VBCSCompiler.exe"
```

Comment:
I see, so `ngenApplication` overrides the default application context, which is VS. So in the change the build task will be ngen'd in the VS context. This seems right. In theory we could also do it in the MSBuild context, but I suspect the VS one is a lot more perf-sensitive.

Reply:
Correct. Currently msbuild itself uses VS context. I think it would be best if the msbuild binaries that may be loaded to devenv process were copied to a different directory than the binaries used by msbuild.exe process. 

https://github.com/Microsoft/msbuild/issues/3377